feat(ImageBlock): allow attachments when block is empty

fulopdaniel opened the pull request:
https://www.figma.com/file/pYt8ECuB6Sc63cVtpKiDgf/Blocks---Updated?type=design&node-id=485-128059&mode=design&t=Ax3uS4Y06OpRKh6h-0

https://www.loom.com/share/911b7fb9330b4b9a8ac98dd2aae8a03f

CU-86947yy57

https://app.clickup.com/t/86947yy57



## Changes

1. Attachments moved to parent component so overflow:hidden is not applied to that.
2. Added a 44px min height to the image component if there are attachments.
3. Hide captions if there is none in view mode.

ragi96 commented on packages/image-block/src/components/ImageCaption.tsx:
```diff
@@ -9,7 +9,7 @@ import {
 import { getCaptionPlugins, titlePlugins } from './helpers';
 import { AppBridgeBlock } from '@frontify/app-bridge';
-import { CaptionPosition, type Settings } from '../types';
+import { CaptionPosition, Ratio, type Settings, textRatioValues } from '../types';
```
```suggestion
import { CaptionPosition, type Ratio, type Settings, textRatioValues } from '../types';
```

fulopdaniel replied:
fixed

ragi96 reviewed (commented):
Also add tests at least for these cases:

1. Attachment set - no asset -> Attachment Shown
2. Attachment set - no asset - download active -> Attachment Shown without Download Button

fulopdaniel commented:
> ![Screenshot 2024-04-05 at 13 49 01](https://private-user-images.githubusercontent.com/9362990/319984893-0184d907-429d-4ef9-b774-fc7d9f056ca4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NTc5MTQsIm5iZiI6MTcxMjU1NzYxNCwicGF0aCI6Ii85MzYyOTkwLzMxOTk4NDg5My0wMTg0ZDkwNy00MjlkLTRlZjktYjc3NC1mYzdkOWYwNTZjYTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDhUMDYyNjU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDM3Njc2ZmVjMTc0N2ZmNzAyN2JlYjRlYjdjOTdhMjZjMWRjMTM2MzFhM2VkMmU0NTM3ZjdkOWY3MDdiYTJiZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.rOtIn9vleHnE_RaNNqxVbBJldXaSdYVMmj9wI68-_rE)
> 
> Displays the download button without an asset

Nice catch, fixed, thanks! Added a testcase for it. (Merged the two test case suggestions you made since they are essentially the same: hide download, show attachments in view mode.)

fulopdaniel commented on packages/image-block/src/components/ImageCaption.tsx:
```diff
@@ -30,36 +31,47 @@ export const ImageCaption = ({
-    return (
+    const hasTitle = hasRichTextValue(name);
+    const hasDescription = hasRichTextValue(description);
+    return isEditing || hasTitle || hasDescription ? (
```
Sure, adjusted

ragi96 reviewed (approved):
LGTM
